lib/getBinaryGltf.js

```diff
@@ -57,57 +58,59 @@ function updateBinaryObject(gltf, pipelineExtras, name, state) {
+        var currentOffset = buffers.binary_glTF.byteLength;
+        var pipelineExtras = buffers.binary_glTF.extras._pipeline;
+        var state = {
+            offset: currentOffset,
+            currentBinaryView: 0
+        };
+
+        if (embed) {
+            updateBinaryObject(gltf, pipelineExtras, 'shaders', state);
+        }
+        if (embedImage) {
+            updateBinaryObject(gltf, pipelineExtras, 'images', state);
+        }
 
-    var body = buffers.binary_glTF.extras._pipeline.source;
-    buffers.binary_glTF.byteLength = state.offset;
-
-    // Remove extras objects before writing
-    removePipelineExtras(gltf);
-
-    // Create padded binary scene string and calculate total length
-    var sceneString = JSON.stringify(gltf);
-    var sceneLength = Buffer.byteLength(sceneString);
-    sceneLength += 4 - (sceneLength % 4);
-    var padding = new Array(sceneLength + 1).join(' ');
-    sceneString = (sceneString + padding).substring(0, sceneLength);
-    var bodyOffset = 20 + sceneLength;
-    var glbLength = bodyOffset + body.length;
-
-    // Write binary glTF header (magic, version, length, sceneLength, sceneFormat)
-    var header = new Buffer(20);
-    header.write('glTF', 0);
-    header.writeUInt32LE(1, 4);
-    header.writeUInt32LE(glbLength, 8);
-    header.writeUInt32LE(sceneLength, 12);
-    header.writeUInt32LE(0, 16);
-
-    // Create scene buffer and overall buffer
-    var scene = new Buffer(sceneString);
-    var glb = Buffer.concat([header, scene, body], glbLength);
-    
-    if (callback) {
-        process.nextTick(function () {
-            callback(header, scene, body);
-        });
-    }
-    return glb;
+        var body = buffers.binary_glTF.extras._pipeline.source;
+        buffers.binary_glTF.byteLength = state.offset;
+
+        // Remove extras objects before writing
+        removePipelineExtras(gltf);
+
+        // Create padded binary scene string and calculate total length
+        var sceneString = JSON.stringify(gltf);
+        var sceneLength = Buffer.byteLength(sceneString);
+        sceneLength += 4 - (sceneLength % 4);
+        var padding = new Array(sceneLength + 1).join(' ');
+        sceneString = (sceneString + padding).substring(0, sceneLength);
+        var bodyOffset = 20 + sceneLength;
+        var glbLength = bodyOffset + body.length;
+
+        // Write binary glTF header (magic, version, length, sceneLength, sceneFormat)
+        var header = new Buffer(20);
+        header.write('glTF', 0);
+        header.writeUInt32LE(1, 4);
+        header.writeUInt32LE(glbLength, 8);
+        header.writeUInt32LE(sceneLength, 12);
+        header.writeUInt32LE(0, 16);
+
+        // Create scene buffer and overall buffer
+        var scene = new Buffer(sceneString);
+        var glb = Buffer.concat([header, scene, body], glbLength);
+
+        if (callback) {
+            process.nextTick(function () {
+                callback(header, scene, body);
```

Comment:
Actually, are the `header` `scene` and `body` parameters necessary? We may be able to take them out entirely.